create_lmodsitepackage.py

```diff
@@ -19,6 +19,32 @@
     return content
 end
 
+local function from_eessi_prefix(t)
+    -- eessi_prefix is the prefix with official EESSI modules
+    -- e.g. /cvmfs/software.eessi.io/versions/2023.06
+    local eessi_prefix = os.getenv("EESSI_PREFIX")
+
+    -- NOTE: exact paths for site and user extensions aren't final, so may need to be updated later.
+    -- See https://github.com/EESSI/software-layer/pull/371
+
+    -- eessi_prefix_host_injections is the prefix with site-extensions (i.e. additional modules)
+    -- to the official EESSI modules, e.g. /cvmfs/software.eessi.io/host_injections/2023.06
+    local eessi_prefix_host_injections = string.gsub(eessi_prefix, 'versions', 'host_injections')
+    -- eessi_prefix_user_home is the prefix with user-extensions (i.e. additional modules)
+    -- to the offocial EESSI modules, e.g. $HOME/eessi/versions/2023.06
+    local eessi_prefix_user_home = string.gsub(eessi_prefix, os.getenv("EESSI_CVMFS_REPO"), pathJoin(os.getenv("HOME"), "eessi"))
+    -- If EESSI_PREFIX wasn't defined, we cannot check if this module was from the EESSI environment
+    -- In that case, we assume it isn't, otherwise EESSI_PREFIX would (probably) have been set
+    if eessi_prefix == nil then
+        return False
+    else
+        -- Check if the full modulepath starts with the eessi_prefix_*
+        return string.find(t.fn, "^" .. eessi_prefix) ~= nil or
+        string.find(t.fn, "^" .. eessi_prefix_host_injections) ~= nil or
+        string.find(t.fn, "^" .. eessi_prefix_user_home) ~= nil
+    end
```

Comment:
```suggestion
    -- If EESSI_PREFIX wasn't defined, we cannot check if this module was from the EESSI environment
    -- In that case, we assume it isn't, otherwise EESSI_PREFIX would (probably) have been set
    if eessi_prefix == nil then
        return False
    else
        -- NOTE: exact paths for site and user extensions aren't final, so may need to be updated later.
        -- See https://github.com/EESSI/software-layer/pull/371

        -- eessi_prefix_host_injections is the prefix with site-extensions (i.e. additional modules)
        -- to the official EESSI modules, e.g. /cvmfs/software.eessi.io/host_injections/2023.06
        local eessi_prefix_host_injections = string.gsub(eessi_prefix, 'versions', 'host_injections')
        
        -- eessi_prefix_user_home is the prefix with user-extensions (i.e. additional modules)
        -- to the offocial EESSI modules, e.g. $HOME/eessi/versions/2023.06
        local eessi_prefix_user_home = string.gsub(eessi_prefix, os.getenv("EESSI_CVMFS_REPO"), pathJoin(os.getenv("HOME"), "eessi"))
        
        -- Check if the full modulepath starts with the eessi_prefix_*
        return string.find(t.fn, "^" .. eessi_prefix) ~= nil or
        string.find(t.fn, "^" .. eessi_prefix_host_injections) ~= nil or
        string.find(t.fn, "^" .. eessi_prefix_user_home) ~= nil
    end
```

Reply:
Urgh, this doesn't look very pretty, but the idea was to just move all the additional paths inside the else statement (since they make no sense if `eessi_prefix` is `nil`)

Reply:
It also removes the  user installations option, this is to reflect changes in the #371 which now allows for user and project installations in arbitrary locations.

Reply:
Ok, this makes sense to me.